### PR TITLE
Added utils crate with newtype macro

### PR DESCRIPTION
**Pull request by royster57**
A convenience macro for implementing the newtype pattern. I'm using it in the mempool and have seen other places where it could be relevant.

### CODE REVIEW

**Review by muursh — APPROVED** [review]
I'm okay with this. A part of me wonders if it's worth deriving the obvious traits in `macro_rules! newtype_derive {}`

File: utils/src/newtype.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#[macro_export]
+macro_rules! newtype {
+    ($vis:vis $name:ident($wrapped:ty)) => {
```
Comment (iljakuklic) [inline utils/src/newtype.rs:3]: I wonder if that would be more idiomatic to more closely follow the usual Rust syntax inside the macro. I.e. the syntax to match would be `$vis struct $name($wrapped)` (notice the struct keyword). Maybe even consider a (optional?) semicolon at the end.

Comment (iljakuklic) [reply utils/src/newtype.rs:3]: Btw does the `$vis:vis` pattern match empty (i.e. private) visibility?

File: utils/src/newtype.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        $vis struct $name($wrapped);
```
Comment (iljakuklic) [inline utils/src/newtype.rs:4]: As per @muursh's comment, I think the macro rule should also match a list of attibutes and expand them before the struct declaration, so we can pass in `#[derive(Blah)]` and whatnot.

**Review by iljakuklic — APPROVED** [review]
I'm fine with this. Added some improvement suggestions as code comments.

**Pull request merged**